### DISCUSSION TITLE
Binding nodes of ui.tree

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#2770] [Ideas / Feature Requests]
Hello everyone,

Would it be possible to bind nodes of `ui.tree` element? Currently I use code such as:

```py
def remove_application_from_tree(tree: ui.tree, app: Application):
    # remove the application from the tree
    nodes = tree._props['nodes']
    for parent_node in nodes:
        parent_node['children'] = [child for child in parent_node['children'] if child['id'] != app.id]
    nodes = [node for node in nodes if node['children']]
    tree._props['nodes'] = nodes
    tree.update()
```

It works and keeps expanded/collapsed state of branches. It would be nice to support methods like `add_node(parent_node_id, node)`, `remove_node(node_id)` and so on.

**Comment by falkoschindler** [comment]
Hi @me21,

I'm not sure how _binding_ nodes could work, but methods like `add_node` and `remove_node` would be quite handy.
Would anyone like to create a pull request? 🙂
